### ISSUE TITLE
Database error finding user (when creating one), supabase-auth accessing public.users not auth.users

### ISSUE THREAD

**Issue by sroussey** [supabase/supabase#1479]
# Bug report

## Describe the bug

I setup local dev a while ago, and now implementing auth. However when I call `supabaseClient.auth.signUp(...)` and look at the results in the net panel of dev tools, I see a result code error 500 and contents `{code: 500, msg: "Database error finding user", ...}` when posting to `http://localhost:8000/auth/v1/signup`

![image](https://user-images.githubusercontent.com/127349/117514139-09b07f00-af48-11eb-80ac-d6b9171f18aa.png)

## System information

- OS: MacOS (latest)
- Browser: Chrome (latest)
- Version of supabase-js: 1.11.9
- Version of Node.js: v14.16.1

**Comment by sroussey** [comment]
Updated to @supabase/supabase-js@1.11.10 but no change.

**Comment by sroussey** [comment]
BTW: how to we update the local supabase docker suite? The `supabase` command line does not have an update command.

**Comment by sroussey** [comment]
Note: my postgress for this is on a different port as i have a lot of local projects. FYI.

**Comment by sroussey** [comment]
Looking in the auth docker log:

```
time="2021-05-08T18:29:10Z" level=error msg="500: Database error finding user" component=api error="error finding user: ERROR: column users.aud does not exist (SQLSTATE 42703)" method=POST path=/token referer="http://localhost:4000/" remote_addr="172.20.0.2:53928" request_id=86482026-2b48-48fb-9197-a2fc00f06e93
```

Which is weird, as I can see auth.users.aud in the db. I do have a public.users that does not have it however.

**Comment by sroussey** [comment]
OK, that is the issue. The auth code is asking for `users` not `auth.users` and so it is seeing the public.users. I have this because I am trying to run the sample https://github.com/vercel/nextjs-subscription-payments

**Comment by MaxAltena** [comment]
It seems that when I use the example from the documentation [for creating users in public.users table from auth.users on create](https://supabase.io/docs/guides/auth#create-a-publicusers-table) I get a similar error.

## Steps
- Follow example from documentation
- Invite user through `Authentication` tab inside supabase
- Receive the following error: Inviting user failed: Database error saving new user.

Might this be somewhat relatable?

**Comment by jasonho-lynx** [comment]
@sroussey did you manage to get `supabase.auth` to point to the `auth.users` table instead?

**Comment by stevenroussey-privicy** [comment]
No. I had to rename the public.users to public.accounts.

**Comment by jasonho-lynx** [comment]
Alright. I did manage to find a temporary workaround by rearranging the entries in [`search_path`](https://github.com/supabase/supabase/blob/master/docker/dockerfiles/postgres/auth-schema.sql#L91) to let `auth` come first before `public`. 

Ideally `supabase.auth` should only focus on the `auth.users` table instead of relying on the search path, but I haven't managed to find the section that specifies it in [gotrue-js](https://github.com/supabase/gotrue-js) yet.

**Comment by jasonho-lynx** [comment]
> It seems that when I use the example from the documentation [for creating users in public.users table from auth.users on create](https://supabase.io/docs/guides/auth#create-a-publicusers-table) I get a similar error.
> 
> ## Steps
> * Follow example from documentation
> * Invite user through `Authentication` tab inside supabase
> * Receive the following error: Inviting user failed: Database error saving new user.
> 
> Might this be somewhat relatable?

@MaxAltena I'm not sure that it's the same since you're using the Supabase hosted version. Your problem sounds more similar to the one specified [here](https://github.com/supabase/supabase/issues/563).

**Comment by jasonho-lynx** [comment]
@sroussey FYI using the [latest dockerfiles](https://github.com/supabase/supabase/tree/master/docker/dockerfiles) seemed to fix the issue for me

**Comment by stevenroussey-privicy** [comment]
How do you upgrade those?

**Comment by jasonho-lynx** [comment]
> How do you upgrade those?

Apologies, I realised even the new version still did not work right away (deleted my comment). However, my earlier comment about editing `search_path` (see the attached link there) still works.

**Comment by wiwatsrt** [comment]
> > How do you upgrade those?
> 
> Apologies, I realised even the new version still did not work right away (deleted my comment). However, my earlier comment about editing `search_path` (see the attached link there) still works.
> 
> tldr: on this [line](https://github.com/supabase/supabase/blob/master/docker/dockerfiles/postgres/auth-schema.sql#L91)
> 
> change `ALTER ROLE postgres SET search_path = "$user", public, auth;`
> 
> to `ALTER ROLE postgres SET search_path = "$user", auth, public;`
> 
> Note that this will have implications on your db's searching behaviour as described in the [postgres docs](https://www.postgresql.org/docs/current/ddl-schemas.html#DDL-SCHEMAS-PATH)

It's Work!
Thank You.

**Comment by madeleineostoja** [comment]
I'm still experiencing this issue, and I don't think editing the auth schema of the docker container is a very good fix. Are there any other workarounds / is this being worked on?

**Comment by drewbietron** [comment]
I'm experiencing this with version `2.7.1` of the js client.  Strange to track down as it happens when I seed my database locally with users.  The first one throws this error and all subsequent signUp calls work fine?

**Comment by tmcw** [comment]
Can we reopen this issue - still seeing this locally, and there isn't a linked commit fixing this, it looks like the accepted solution is a workaround.

**Comment by aavetis** [comment]
still an issue, and this doesn't feel like a nice to have. the database in prod works fine having auth.users and public.users; just doesn't work locally

**Comment by tholder** [comment]
I'm seeing this, strangely when running a test rather than actually using my app normally. I don't have a public.users table.

**Comment by joshmarnold** [comment]
I'm also see this

**Comment by Vedsaga** [comment]
Still getting error locally... tried `supabase db reset` and then created few schema and now surrenly getting error... before DB-Rest it was working fine...
```
AuthException(message: Database error finding user, statusCode: 500)
```

**Comment by Vedsaga** [comment]
The issues get's solved after delete all old docker images and running `npx supabase start` to pulling new image from scratch...

**Comment by benaduggan** [comment]
I was running into this, and my issue turned out being that the associated identity in `auth.identities` had null values in created_at and updated_at which caused the Go code to die resulting in this log

```
{"component":"api","error":"error finding user: unable to fetch records: sql: Scan error on column index 0, name \"created_at\": unsupported Scan, storing driver.Value type \u003cnil\u003e into type *time.Time","level":"error","method":"POST","msg":"500: Database error finding user","path":"/admin/generate_link","referer":"https://app.magicschool.ai","remote_addr":"2a06:98c0:3600::103","request_id":"90b7ab83c3d7944d-SJC","time":"2025-02-02T05:09:35Z"}
```

setting both created_at and updated_at to a timestamp fixed the issue

**Comment by Hallidayo** [comment]
Hi all, Just checking in here. Is this still an issue on recent versions?

**Comment by Jusharra** [comment]
I'm seeing this issue occur no matter what tI try to fix it. Was there ea fix? I attempted some of the solutions mentioned but I'm still seeing the same error.
